### DISCUSSION TITLE
Processing a soap request to a webservice listener containing elements with mustUnderstand attributes in the header

### DISCUSSION THREAD

**Discussion by THendriksDecos** [frankframework/frankframework#6398] [Q&A]
I am trying to set up a webservice endpoint but I am having difficulties having the request get accepted into the pipeline. This is due to two elements with mustUnderstand attribute in the soap header: A To and Action element, see request format example below.

I tried to achieve this using a soapinputvalidator (as well as with an xmlvalidator and a WSDLValidator) but was unable to have the request enter the pipeline. The request is send to an instance of a local Frank!Runner using SoapUI 5.7.2

How can the soap headers be processed so that the soap headers are "understood" using Frank Framework?

Example request format: 
```
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <soap:Header>
    <To soap:mustUnderstand="1" xmlns="http://schemas.microsoft.com/ws/2005/05/addressing/none">{TargetURL}</To>
    <Action soap:mustUnderstand="1" xmlns="http://schemas.microsoft.com/ws/2005/05/addressing/none">{SOAPACTION}</Action>
  </soap:Header>
  <soap:Body>...
```
Response: 
```
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
	<soap:Body>
		<soap:Fault>
			<faultcode>soap:MustUnderstand</faultcode>
			<faultstring>MustUnderstand headers: [{http://schemas.microsoft.com/ws/2005/05/addressing/none}To, {http://schemas.microsoft.com/ws/2005/05/addressing/none}Action] are not understood.</faultstring>
		</soap:Fault>
	</soap:Body>
</soap:Envelope>
```

**Comment by jjansenvr** [comment]
Currently we do not support must understand on headers, the default in soap implementations is that it returns an error if you add it. 
To fix this we would need to implement a MustUnderstandInterceptor in our CXF implementation.

what is your functional expectation of using this property on this header?

https://github.com/apache/cxf/blob/main/rt/bindings/soap/src/main/java/org/apache/cxf/binding/soap/interceptor/MustUnderstandInterceptor.java
